raiden-network/light-client
# Withdraw

SDK part of #249 

**Short description**
Adds withdraw functionality to SDK.
Call it with
```ts
await raiden.withdrawChannel('0xtoken', '0xpartner', '100');
```
It returns a Promise which resolves to the `txHash` of the withdraw transaction. The third param is optional. If left undefined, it withdraws the available capacity. Otherwise, it should be a value between 1 and the maximum withdrawable amount, which is now also exposed in the `RaidenChannel` objects in `channels$: Observable<RaidenChannels>` as a `ownWithdrawable` property. The `RaidenChannel` interface was also expanded to include several amounts related to the channel. See `ChannelBalances` for a list.

**Definition of Done**


**Steps to manually test the change (dApp)**

1. Call the `withdrawChannel` public method on a channel with an online partner
2. See the messages flowing, and finally the `setTotalWithdraw` transaction prompt to perform the withdraw on-chain
